.github/workflows/python-poetry-tests.yml

```diff
@@ -23,6 +28,7 @@ jobs:
   test:
     name: 🔬 Test (🐍 ${{ inputs.python-version }})
     runs-on: ubuntu-latest
+    container: ${{ inputs.container-image }}
```

Comment:
What is happening if `container-image` argument is not provided?

Reply:
I believe you might want to add also the `if` condition of https://github.com/orgs/community/discussions/26756#discussioncomment-3253265, i.e.:
```yaml
container:
  image: ${{ inputs.container-image }}
  if: # something like `${{ inputs.container-image }} != null`
```

Reply:
If `container-image` is not provided, then the workflow works perfectly fine as usual. For some reasons, the `container` property is allowed to be an empty string but the `image` property not.

The only thing I need to figure out now is to best consider the case where the input is an object (either using `matrix` or a different thing).

Reply:
Why do you want to allow the input to be an object? :upside_down_face:

Reply:
> Why do you want to allow the input to be an object? upside_down_face

Ah, right! The container registry can simply be passed to the container property without the need of `image`.

Reply:
> I believe you might want to add also the `if` condition of https://github.com/orgs/community/discussions/26756#discussioncomment-3253265, i.e.:
> 
> ```yaml
> container:
>   image: ${{ inputs.container-image }}
>   if: # something like `${{ inputs.container-image }} != null`
> ```

I don't think the `if` statement is needed anymore here. I tried omitting the `container-image` input and it works as expected. So, for sure, if you test this with eko/yadism it will no longer raises an error.

Reply:
At this point it is a sufficiently simple and minimal change. I would just merge, and roll back in case it is not working.